harness/tests/integration_cases/test_raft.rs

```diff
@@ -2441,6 +2441,34 @@ fn test_read_only_for_new_leader() {
     assert_eq!(rs.request_ctx, wctx.as_bytes().to_vec());
 }
 
+// `test_advance_commit_index_by_read_index_response` ensures that read index response
+// can advance the follower's commit index if it has new enough logs
+#[test]
+fn test_advance_commit_index_by_read_index_response() {
+    let l = default_logger();
+    let mut tt = Network::new(vec![None, None, None, None, None], &l);
+    tt.send(vec![new_message(1, 1, MessageType::MsgHup, 0)]);
+
+    // don't commit entries
+    tt.cut(1, 3);
+    tt.cut(1, 4);
+    tt.cut(1, 5);
+    tt.send(vec![new_message(1, 1, MessageType::MsgPropose, 1)]);
+    tt.send(vec![new_message(1, 1, MessageType::MsgPropose, 1)]);
+
+    tt.recover();
+    tt.cut(1, 2);
+
+    // commit entries for leader but not node 2
+    tt.send(vec![new_message(3, 1, MessageType::MsgReadIndex, 1)]);
+    assert_eq!(tt.peers[&1].raft_log.committed, 4);
+    assert_eq!(tt.peers[&2].raft_log.committed, 2);
+
+    tt.recover();
+    tt.send(vec![new_message(2, 1, MessageType::MsgReadIndex, 1)]);
```

Comment:
Should make leader use lease-base read option, otherwise heartbeat is still sent out.

Reply:
Fixed